nano/nano_node/entry.cpp

```diff
@@ -142,1894 +140,1870 @@ int main (int argc, char * const * argv)
-			if (pow_sleep_interval_it != vm.cend ())
+		nano::work_pool work (std::numeric_limits<unsigned>::max (), pow_rate_limiter);
+		nano::change_block block (0, 0, nano::keypair ().prv, 0, 0);
+
+		std::cerr << boost::str (boost::format ("Starting generation profiling. Difficulty: %1$#x (%2%x from base difficulty %3$#x)\n") % difficulty % nano::to_string (nano::difficulty::to_multiplier (difficulty, network_constants.publish_full.base), 4) % network_constants.publish_full.base);
+		while (true)
+		{
+			block.hashables.previous.qwords[0] += 1;
+			auto begin1 (std::chrono::high_resolution_clock::now ());
+			block.block_work_set (*work.generate (nano::work_version::work_1, block.root (), difficulty));
+			auto end1 (std::chrono::high_resolution_clock::now ());
+			std::cerr << boost::str (boost::format ("%|1$ 12d|\n") % std::chrono::duration_cast<std::chrono::microseconds> (end1 - begin1).count ());
+		}
+	}
+	else if (vm.count ("debug_profile_validate"))
+	{
+		uint64_t difficulty{ nano::network_constants ().publish_full.base };
+		std::cerr << "Starting validation profile" << std::endl;
+		auto start (std::chrono::steady_clock::now ());
+		bool valid{ false };
+		nano::block_hash hash{ 0 };
+		uint64_t count{ 10000000U }; // 10M
+		for (uint64_t i (0); i < count; ++i)
+		{
+			valid = nano::work_v1::value (hash, i) > difficulty;
+		}
+		std::ostringstream oss (valid ? "true" : "false"); // IO forces compiler to not dismiss the variable
+		auto total_time (std::chrono::duration_cast<std::chrono::nanoseconds> (std::chrono::steady_clock::now () - start).count ());
+		uint64_t average (total_time / count);
+		std::cout << "Average validation time: " << std::to_string (average) << " ns (" << std::to_string (static_cast<unsigned> (count * 1e9 / total_time)) << " validations/s)" << std::endl;
+	}
+	else if (vm.count ("debug_opencl"))
+	{
+		nano::network_constants network_constants;
+		bool error (false);
+		nano::opencl_environment environment (error);
+		if (error)
+		{
+			std::cout << "Error initializing OpenCL" << std::endl;
+			return -1;
+		}
+
+		unsigned short platform (0);
+		auto platform_it = vm.find ("platform");
+		if (platform_it != vm.end ())
+		{
+			try
 			{
-				pow_rate_limiter = std::chrono::nanoseconds (boost::lexical_cast<uint64_t> (pow_sleep_interval_it->second.as<std::string> ()));
+				platform = boost::lexical_cast<unsigned short> (platform_it->second.as<std::string> ());
 			}
-
-			nano::work_pool work (std::numeric_limits<unsigned>::max (), pow_rate_limiter);
-			nano::change_block block (0, 0, nano::keypair ().prv, 0, 0);
-			if (!result)
+			catch (boost::bad_lexical_cast &)
 			{
-				std::cerr << boost::str (boost::format ("Starting generation profiling. Difficulty: %1$#x (%2%x from base difficulty %3$#x)\n") % difficulty % nano::to_string (nano::difficulty::to_multiplier (difficulty, network_constants.publish_full.base), 4) % network_constants.publish_full.base);
-				while (!result)
```

Comment:
This `while(!result)` and the previous `if (!result)` didn't make much sense because the code guaranteed `result` would be untouched and thus equal to `0`. So I just ignored the `if` and added a `while (true)` instead.